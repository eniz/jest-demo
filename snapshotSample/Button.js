/** @format */
/**
 * External dependencies
 */
import React, { PureComponent } from 'react';
import classNames from 'classnames';

export default class Button extends PureComponent {
  render() {
    const className = classNames( 'button', this.props.className, {
      'is-compact': this.props.compact,
      'is-primary': this.props.primary,
      'is-scary': this.props.scary,
      'is-busy': this.props.busy,
      'is-borderless': this.props.borderless,
    } );

    if ( this.props.href ) {
      const { compact, primary, scary, busy, borderless, type, ...props } = this.props;

      // block referrers when external link
      const rel = props.target
        ? ( props.rel || '' ).replace( /noopener|noreferrer/g, '' ) + ' noopener noreferrer'
        : props.rel;

      return <a { ...props } rel={ rel } className={ className } />;
    }

    const { compact, primary, scary, busy, borderless, target, rel, ...props } = this.props;

    return <button { ...props } className={ className } />;
  }
}

Button.defaultProps = {
  type: 'button'
}
