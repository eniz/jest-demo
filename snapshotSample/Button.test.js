import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';


it('renders correctly a tag', () => {
  const tree = renderer
    .create(<Button href="https://www.otel.com">Otel.com</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders correctly button tag', () => {
  const tree = renderer
    .create(<Button type="submit">Send</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
