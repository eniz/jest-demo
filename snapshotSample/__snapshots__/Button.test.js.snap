// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a tag 1`] = `
<a
  className="button"
  href="https://www.otel.com"
>
  Otel.com
</a>
`;

exports[`renders correctly button tag 1`] = `
<button
  className="button"
  type="submit"
>
  Send
</button>
`;
